packages/scripts/src/components/front-end/payments/paypal/buttons.ts

```diff
@@ -0,0 +1,334 @@
+  } catch (error) {
+    logError('Network error during order capture:', error);
+    return null;
+  }
+}
+
+// Load PayPal SDK dynamically
+function loadPayPalSDK(clientId: string, sandbox: boolean, currency: string = 'USD'): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if ((window as Window & { paypal?: PayPalSDK }).paypal) {
+      resolve();
+      return;
+    }
+
+    // Check if script is already being loaded
+    const existingScript = document.getElementById('freeform-paypal-sdk');
+    if (existingScript) {
+      existingScript.addEventListener('load', () => resolve());
+      existingScript.addEventListener('error', () => reject(new Error('Failed to load PayPal SDK')));
+      return;
+    }
+
+    const script = document.createElement('script');
+    script.id = 'freeform-paypal-sdk';
+    const cur = (currency || 'USD').toUpperCase();
+    script.src = `https://www.paypal.com/sdk/js?client-id=${clientId}&currency=${cur}&intent=capture&components=buttons&enable-funding=venmo,paylater`;
+    script.async = true;
+    script.defer = true;
+
+    script.onload = () => {
+      // SDK loaded
+      resolve();
+    };
+
+    script.onerror = () => {
+      logError('Failed to load SDK');
+      reject(new Error('Failed to load PayPal SDK'));
+    };
+
+    document.head.appendChild(script);
+  });
+}
+
+// Initialize PayPal buttons for a specific element
+async function initializePayPalButtons(root: Element) {
+  const config = parseConfig(root);
+  if (!config) {
+    logError('Could not parse config from element', root);
+    return;
+  }
+
+  if (!config.clientId) {
+    logError('No client ID provided');
+    return;
+  }
+
+  const form = root.closest('form');
+  if (!form) {
+    logError('Could not find form element');
+    return;
+  }
+
+  const hiddenInput = form.querySelector<HTMLInputElement>(Q.orderInput);
+  if (!hiddenInput) {
+    logError('Could not find hidden order input');
+    return;
+  }
+
+  // Initialize PayPal buttons
+  try {
+    // Load SDK first
+    await loadPayPalSDK(config.clientId, config.sandbox, config.currency || 'USD');
+
+    // Wait a bit for the SDK to be fully ready
+    await new Promise((resolve) => setTimeout(resolve, 100));
+
+    const p = (window as unknown as { paypal?: PayPalSDK }).paypal;
+    if (!p) {
+      logError('PayPal SDK is not available on window');
+      return;
+    }
+
+    p.Buttons({
+      style: {
+        layout: 'vertical',
+        color: 'blue',
+        shape: 'rect',
+        label: 'paypal',
+      },
+      createOrder: async () => {
+        // Create order
+
+        try {
+          // Serialize current form values so server can compute dynamic amount before order create
+          const fd = new FormData(form);
+          const values: Values = {};
+          fd.forEach((v, k) => {
+            (values as Record<string, unknown>)[k] = v as unknown as string | number | boolean | File;
+          });
```

Comment:
You have some values typed already and don't need to re-type them here.

```suggestion
          const formData = new FormData(form);
          const values: Values = {};
          formData.forEach((value, key) => {
            values[key] = value;
          });
```